linalg/solvers.hpp

```diff
@@ -799,6 +800,61 @@ class KLUSolver : public Solver
 
 #endif // MFEM_USE_SUITESPARSE
 
+/// Block diagonal solver for A, each block is inverted by direct solver
+class BlockDiagSolver : public Solver
```

Comment:
I think this needs a more descriptive name. MFEM already has `BlockDiagonalPreconditioner` and this object needs to be clearly different. Maybe `BlockDiagonalDirectSolver` or `DirectSubBlockSolver`?